packages/components/src/internal/components/domainproperties/assay/AssayPropertiesPanel.tsx

```diff
@@ -34,251 +36,209 @@ import {
+            {!appPropertiesOnly && hasModule('study', moduleContext) && (
                 <Col xs={12} lg={6}>
-                    {!appPropertiesOnly && this.renderLinkToStudySettings()}
+                    <div className="domain-field-padding-bottom">
+                        <SectionHeading title="Link to Study Settings" />
+                        <AutoLinkDataInput model={model} onChange={onInputChange} />
+                        <AutoLinkCategoryInput model={model} onChange={onInputChange} />
+                    </div>
                 </Col>
-            </Form>
-        );
-    }
+            )}
+        </Form>
+    );
+});
 
-    renderPanel() {
-        const { model, helpTopic } = this.props;
-        const { isValid } = this.state;
+AssayPropertiesForm.displayName = 'AssayPropertiesForm';
 
+interface OwnProps extends AssayPropertiesFormProps {
+    asPanel?: boolean;
+    helpTopic?: string;
+}
+
+type Props = OwnProps & BasePropertiesPanelProps;
+
+const AssayPropertiesPanelImpl: FC<Props & InjectedDomainPropertiesPanelCollapseProps> = memo(props => {
+    const { appPropertiesOnly, asPanel, children, helpTopic, model, onChange } = props;
+    const [isValid, setIsValid] = useState<boolean>(true);
+
+    const updateValidStatus = useCallback(
+        (newModel?: AssayProtocolModel): void => {
+            const updatedModel = newModel || model;
+            setIsValid(!!updatedModel?.hasValidProperties());
+
+            // Issue 39918: only consider the model changed if there is a newModel param
+            if (newModel) {
+                onChange(updatedModel);
+            }
+        },
+        [model, onChange]
+    );
+
+    const form = (
+        <AssayPropertiesForm appPropertiesOnly={appPropertiesOnly} model={model} onChange={updateValidStatus}>
+            {children}
+        </AssayPropertiesForm>
+    );
+
+    if (asPanel) {
         return (
             <BasePropertiesPanel
-                {...this.props}
+                {...props}
                 headerId={PROPERTIES_HEADER_ID}
-                title="Assay Properties"
-                updateValidStatus={this.updateValidStatus}
                 isValid={isValid}
+                title="Assay Properties"
+                updateValidStatus={updateValidStatus}
             >
                 {helpTopic && <HelpTopicURL nounPlural="assays" helpTopic={helpTopic} />}
-                {this.renderForm()}
+                {form}
             </BasePropertiesPanel>
         );
     }
 
-    render() {
-        return <>{this.props.asPanel ? this.renderPanel() : this.renderForm()}</>;
-    }
-}
+    return form;
+});
+
+AssayPropertiesPanelImpl.defaultProps = {
+    appPropertiesOnly: false,
+    asPanel: true,
+    helpTopic: DEFINE_ASSAY_SCHEMA_TOPIC,
+};
+
+AssayPropertiesPanelImpl.displayName = 'AssayPropertiesPanelImpl';
```

Comment:
Thanks for refactoring to a `FC` in this PR as well. Much cleaner and the split with the `AssayPropertiesPanelImpl` and the `AssayPropertiesForm` looks good!